File: MyProject/Source/MyProject/Private/BasicFrame/Libs/FrameHandle/SystemTimeData.cpp
#include "MyProject.h"
#include "SystemTimeData.h"
#include "EngineApi.h"

SystemTimeData::SystemTimeData()
{
	mPreTime = 0;
	mCurTime = 0;
	mDeltaSec = 0.0f;
}

float SystemTimeData::getDeltaSec()
{
	return mDeltaSec;
}

void SystemTimeData::setDeltaSec(float value)
{
	mDeltaSec = value;
}

long SystemTimeData::getCurTime()
{
	return mCurTime;
}
void SystemTimeData::setCurTime(long value)
{
	mCurTime = value;
}

void SystemTimeData::nextFrame()
{
	mCurTime = EngineApi::getUTCSec();
	if (mPreTime != 0.0f)     // 第一帧跳过，因为这一帧不好计算间隔
	{
		mDeltaSec = mCurTime - mPreTime;
	}
	else
	{
		mDeltaSec = 1 / 24;        // 每秒 24 帧
	}
	mPreTime = mCurTime;
}

File: MyProject/Content/MyAsset/Lua/MyLua/Libs/FrameHandle/SystemTimeData.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

local M = GlobalNS.Class(GlobalNS.GObject);
M.clsName = "SystemTimeData";
GlobalNS[M.clsName] = M;

function M:ctor()
    self.mPreTime = 0;              -- 上一次更新时的秒数
    self.mCurTime = 0;              -- 正在获取的时间
    self.mDeltaSec = 0;             -- 两帧之间的间隔
end

function M:getDeltaSec()
    return self.mDeltaSec;
end

function M:setDeltaSec(value)
    self.mDeltaSec = value;
end

function M:getCurTime()
    return self.mCurTime;
end

function M:setCurTime(value)
    self.mCurTime = value;
end

function M:nextFrame()
    self.mPreTime = self.mCurTime;
end

return M;